### ISSUE TITLE
output_as_data_frames() does not include meter_serial_number 

### ISSUE THREAD

**Issue by simon-anz** [aguinane/nem-reader#31]
After #9 the meter_serial_number is not yet included in the returned DataFrame(s)

**Comment by simon-anz** [comment]
related issue: overlapping data when >1 meter not distinguishable

**Comment by aguinane** [comment]
Makes sense, should be able to add a test for that and then work out a solution to fix it.
It merges the channels based on the index which is t_start so might need to make serial no as part of the index as well, not sure how that would work (and might slow it down). Might need to do a check to see if there are 2 meters and handle it differently.

**Comment by lyricnz** [comment]
Here is screenshot of my CSV (happy to provide if it would help), with most of the 300 records hidden. I had a new meter installed on 2022-01-28 (old one turned off 0830, new one turned on sometime that day; then solar generation enabled late 2022-08-15) so another block with negative consumption numbers.

![image](https://user-images.githubusercontent.com/122371/190520641-2566d8b1-d142-453f-af9f-4976db060890.png)

**Comment by aguinane** [comment]
Would it make more sense to return a giant dataframe with all NMIs as well?

**Comment by lyricnz** [comment]
> Would it make more sense to return a giant dataframe with all NMIs as well?

How would you structure it?  Nested-columns (or whatever that's called)?

> TLDR the problem is that the dataframe has an index of t_start and a single column for each channel (so there's no logical place to put the meter_serial_number)

**Comment by aguinane** [comment]
Looks like it could use a [pandas.MultiIndex](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.MultiIndex.html).

**Comment by aguinane** [comment]
v0.8.0 now includes an option to export a dataframe that also includes serial no ` get_pivot_data_frame(split_days=False, set_interval=None, include_serno=False) `
